### DISCUSSION TITLE
Facets at Component Level

### DISCUSSION THREAD

**Discussion by javihern98** [bis-med-it/pysdmx#31] [General]
We need to better specify the Facets inside the Component, specially if they are located in the Core Representation or in the Local Representation. Suggested class:

```python
class Component(Struct, frozen=True, omit_defaults=True):
    id: str
    required: bool
    role: Role
    dtype: DataType = DataType.STRING
    local_representation_facets: Optional[Facets] = None
    name: Optional[str] = None
    description: Optional[str] = None
    codes: Union[Codelist, Hierarchy, None] = None
    attachment_level: Optional[str] = None
    array_def: Optional[ArrayBoundaries] = None
    concept_identity: Optional[Concept] = None

    def __str__(self) -> str:
        """Returns a human-friendly description."""
        out = []
        for k in self.__annotations__.keys():
            v = self.__getattribute__(k)
            if v:
                out.append(f"{k}={str(v)}")
        return ", ".join(out)

    def facets(self) -> Optional[Facets]:
        # TODO: Add here Core Representation Facets also
        return self.local_representation_facets

```

**Comment by sosna** [comment]
Hi Javier, 

I discussed with Stratos and the proposal looks good to us, thanks. Still, we were wondering why you mentioned facets only. I guess the same logic should apply to the codes and the data type, right? After all, they too could be defined either at the concept-level or at the DSD-level. Would you agree?

If so, happy to make the changes.

Let me know :)

Xavier

**Reply by javihern98** [reply]
I agree, still I have not find any issues yet, I would wait to get the metadata XML parsers implemented to see if this is also an issue. We can leave it as-is until then or make the changes you feel comfortable, as you prefer.

**Comment by javihern98** [comment]
I will generate a new issue to implement only this on Component Level
